src/main/java/org/asteriskjava/manager/event/QueueMemberEvent.java

```diff
@@ -93,25 +94,54 @@ public void setQueue(String queue)
     }
 
     /**
-     * Returns the name of the member's interface.
-     * <p>
-     * E.g. the channel name or agent group (for example "Agent/@1").
-     * 
+     * Returns the name of the member's interface.<p>
+     * E.g. the channel name or agent group.
+     *
      * @return the name of the member's interface.
      */
-    public String getLocation()
+    final public String getInterface()
     {
-        return location;
+        return _interface;
     }
 
     /**
      * Sets the name of the member's interface.
      * 
-     * @param location the name of the member's interface.
+     * @param member the name of the member's interface.
+     */
+    final public void setInterface(String _interface)
+    {
+        this._interface = _interface;
+    }
+
+    /**
+     * Returns the name of the member's interface.<p>
+     * E.g. the channel name or agent group.
+     *
+     * @deprecated since Asterisk 12
+     *
+     * @return the name of the member's interface.
+     */
+    @Deprecated
+    final public String getLocation()
+    {
+        return _interface;
+    }
+
+    /**
+     * Sets the name of the member's interface.
+     *
+     * @deprecated since Asterisk 12
+     *
+     * @param member the name of the member's interface.
      */
-    public void setLocation(String location)
+    @Deprecated
+    final public void setLocation(String _interface)
     {
```

Comment:
why do you allow setInterface to set a null interface but not setLocation?

Reply:
setInterface will not be called prior to Asterisk 12. But it seems that asteriskjava calls setLocation with null from asterisk 12 on and can overwrite the correct field already set with setInterface